storage/innobase/handler/handler0alter.cc

```diff
@@ -1261,15 +1262,6 @@ ha_innobase::check_if_supported_inplace_alter(
 				}
 				break;
 			default:
-				/* For any other data type, NULL
-				values are not converted.
-				(An AUTO_INCREMENT attribute cannot
-				be introduced to a column with
-				ALGORITHM=INPLACE.) */
-				ut_ad((MTYP_TYPENR((*af)->unireg_check)
-				       == Field::NEXT_NUMBER)
-				      == (MTYP_TYPENR(f->unireg_check)
-					  == Field::NEXT_NUMBER));
 				goto next_column;
```

Comment:
Please retain the first 2 lines of the comment.

Can you please add some proof that `dict_table_t::persistent_autoinc` is being correctly updated when adding or removing the `AUTO_INCREMENT` attribute of a column? I fear that this could potentially introduce a bug related to that, similar to what was fixed in 10.4 in commit dc6ad597656e8173dd73edb58a967152df70dc9e and commit 88b7b8199a6061ccdeb2e2a1ac789eb98e02ac92.